src/components/UpdateProfile/index.js

```diff
@@ -0,0 +1,172 @@
+  title,
+  firstName,
+  lastName,
+  changePasswordLabel,
+  changePasswordLink,
+  updateLabel,
+  successLabel,
+  errorLabel,
+  ...props
+}) {
+  const classes = useStyles(props);
+  const { data: session } = useSession();
+  const [openSnackBar, setOpenSnackBar] = useState(false);
+  const [message, setMessage] = useState("");
+  const handleSnackBarClose = (event, reason) => {
+    if (reason === "clickaway") {
+      return;
+    }
+    setOpenSnackBar(false);
+  };
+  useEffect(() => {
+    if (!session) {
+      Router.push("/login");
+    }
+  }, [session]);
+  const [form, setForm] = useState({
+    firstName: session.user.first_name,
+    lastName: session.user.last_name,
+  });
+
+  const formChanged = (event) => {
+    setForm({
+      ...form,
+      [event.target.name]: event.target.value,
+    });
+  };
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+
+    try {
+      await fetchJson("/api/update-profile", session, {
+        method: "PATCH",
+        body: JSON.stringify(form),
+      });
+      setOpenSnackBar(true);
+      setMessage({ status: "success", text: successLabel });
+    } catch (e) {
+      setOpenSnackBar(true);
+      setMessage({ status: "error", text: errorLabel });
+    }
+  };
+
+  return (
+    <Section className={classes.section}>
+      <Grid container>
+        <Grid item xs={7} className={classes.container}>
+          <Typography variant="h2">{title}</Typography>
+          <form className={classes.form}>
+            <Grid container>
+              <Grid item xs={12}>
+                <TextField
+                  className={classes.textfield}
+                  InputLabelProps={{ shrink: false, className: classes.label }}
+                  InputProps={{ className: classes.input }}
+                  autoComplete="firstName"
+                  name="firstName"
+                  variant="outlined"
+                  fullWidth
+                  id="firstName"
+                  label={firstName}
+                  autoFocus
+                  value={form.firstName}
+                  onChange={formChanged}
+                />
+              </Grid>
+
+              <Grid item xs={12}>
+                <TextField
+                  className={classes.textfield}
+                  InputLabelProps={{ shrink: false, className: classes.label }}
+                  InputProps={{ className: classes.input }}
+                  autoComplete="lastName"
+                  name="lastName"
+                  variant="outlined"
+                  fullWidth
+                  id="lastName"
+                  label={lastName}
+                  value={form.lastName}
+                  autoFocus
+                  onChange={formChanged}
+                />
+              </Grid>
+              <Grid item xs={12}>
+                <Link className={classes.link} href={changePasswordLink}>
+                  {changePasswordLabel}
+                </Link>
+              </Grid>
```

Comment:
If we don't have functionality for this (since it's mostly OAuth logins for now), lets not add it until we do.